R/florabank_queries.R

```diff
@@ -0,0 +1,371 @@
+#' Query the florabank to get taxon trait values for (a) taxon trait(s)
+#'
+#' This function takes as input (part of) a taxon trait, queries the florabank
+#' and returns the taxon trait values in a tidy data format
+#'
+#' @param trait_name A (part of) a trait name for which you want to get the
+#' associated taxon-specific trait values. If this is missing, the function
+#' returns an error and prints a message showing all possible trait names.
+#'
+#' @return A dataframe containing the trait values for each species and for all
+#' partially matched traits. The dataframe contains the variables TaxonID,
+#' TaxonAfkorting, TaxonWetenschappelijk, TaxonNederlands, Kenmerk, Code,
+#' Omschrijving en Rekenwaarde. The first four variables identify the taxon,
+#' the latter four variables relate to the taxon traits.
+#'
+#' @importFrom DBI dbDisconnect
+#' @importFrom dplyr
+#' tbl
+#' collect
+#' distinct
+#' pull
+#' %>%
+#' inner_join
+#' left_join
+#' filter
+#' select
+#' rename
+#'@importFrom rlang .data
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'  # get all Ellenberg values via partial matching
+#' fb_ellenberg <- florabank_traits("llenberg")
+#' # get all red lists via partial matching
+#' fb_rodelijsten <- florabank_traits("rode")
+#' # get only the red list for vascular plant species
+#' fb_rodelijstvaatplanten <- florabank_traits("Rode lijst Vaatplanten")
+#' #if the trait_name argument is missing, a list of possible names is printed
+#' florabank_traits()
+#' }
+
+florabank_traits <- function(trait_name) {
+  db_connectie <- connect_inbo_dbase("D0021_00_userFlora")
```

Comment:
@hansvancalster I would actually rather use a connection as first input of the functions instead of creating the dbase connection inside the function:
- easier to provide customization on the connection side
- make sure you're not creating multiple dbase connections when using the different functions, but reuse a single connection for all of them.

Reply:
Not sure I agree... for these functions, the connection need not to be customized (it is always a connection to the florabank), but maybe I am overlooking situations (different operating systems?). Not sure if multiple dbase connections is an issue: the connection is also closed within the function.

Reply:
I agree with Stijn. Opening and closing database connection is not good if you do it repeatedly. Add the connection as an argument to the function. Consider using the `pool` package for managing connections. @ElsLommelen has experience using that.

Reply:
No worries, I will change this. The reasons why this is best practice were just not that clear to me.
Thank you both for your thoughts about this.

Reply:
Resolved in 168e608